### PR TITLE
Windows: Upgrade JACK dependency to 1.9.21

### PR DESCRIPTION
**Pull request by ann0see**
**Short description of changes**
Another maintenance upgrading PR. 

CHANGELOG: Windows: Upgraded JACK build to use JACK 1.9.21 

**Context: Fixes an issue?**

No. 

**Does this change need documentation? What needs to be documented and how?**

No.

**Status of this Pull Request**

Should be tested. 

**What is missing until this pull request can be merged?**

Test + CI

## Checklist

-  [x] I've verified that this Pull Request follows the [general code principles](https://github.com/jamulussoftware/jamulus/blob/master/CONTRIBUTING.md#jamulus-projectsource-code-general-principles)
-  [x] I tested my code and it does what I want
-  [x] My code follows the [style guide](https://github.com/jamulussoftware/jamulus/blob/master/CONTRIBUTING.md#source-code-consistency) 
-  [x] I waited some time after this Pull Request was opened and all GitHub checks completed without errors. 
-  [x] I've filled all the content above

### CODE REVIEW

**Comment by hoffie** [comment]
Culprit seems to be here:
https://github.com/jamulussoftware/jamulus/runs/7011426493?check_suite_focus=true#step:7:160

Some wild suggestions:
- Access denial might stem from the fact the installer still runs when choco expects it to have finished. It seems like the AutoHotkey logic was updated as part of this release. Maybe the installer hangs in a non-finished step.
- Access denial might be from Windows Defender (?)

Logs will probably help, not sure if it's worth to spend the effort here now, though?

**Comment by ann0see** [comment]
Hmm. Yes. De-tagged. Probably worth focusing on other stuff first.

**Comment by henkdegroot** [comment]
FWIW, when I try to install manually on a Win2022 server...then the autokey script which should take care of the installation of jack, does not kick in. Sounds like something is not working as expected. I posted a message on chocolatey. Hope to get a reply.

**Comment by henkdegroot** [comment]
The autohotkey script is not working due to changes in the installer screens. Buttons names are slightly different. I have submitted a PR containing a fix to the repo of the package maintainer. Until fixed, we cannot use 1.9.21.

**Comment by hoffie** [comment]
Cool, thanks!
Linking the PR here: https://github.com/chtof/chocolatey-packages/pull/73

File: .github/autobuild/windows.ps1 [hunk 1]
```diff
@@ -19,7 +19,7 @@ $ChocoCacheDir = 'C:\ChocoCache'
 $Qt32Version = "5.15.2"
 $Qt64Version = "5.15.2"
 $AqtinstallVersion = "2.1.0"
-$JackVersion = "1.9.17"
+$JackVersion = "1.9.21"
```
Comment (henkdegroot) [inline .github/autobuild/windows.ps1:22]: @ann0see, the JACK version at choco has been updated. The version to install is now: 1.9.21.20220626
I have validated this on my local system and the auto install works when using that version number.

**Review by henkdegroot — CHANGES_REQUESTED** [review]

File: .github/autobuild/windows.ps1 (hunk 1 above)
Comment (henkdegroot) [reply .github/autobuild/windows.ps1:22]: Hmm...seems I selected the wrong option for submitting my comment.

**Review by hoffie — APPROVED** [review]
This seems to have been forgotten somehow as it lacked a milestone? I think it's good to merge, isn't it?

**Comment by henkdegroot** [comment]
Perhaps another (re) run of the action script? Just to confirm the auto install still works?
Otherwise it should be good to merge.

**Comment by hoffie** [comment]
I don't see a way to trigger a re-run via UI.
@ann0see Can you just rebase and push?

**Comment by ann0see** [comment]
Done

**Comment by hoffie** [comment]
@henkdegroot Can you validate the installation?

**Comment by henkdegroot** [comment]
I downloaded the artifact for Windows JACK and installed on my system. All fine.
My system had JACK v1.9.19 installed and Jamulus works correct. Connecting to server and audio transmits correct.
Next I upgraded my system to JACK v1.9.21 and all still working fine.

Also when JACK is not running, Jamulus detects it and alerts the user.

I would say all good to go.

File: .github/autobuild/windows.ps1 [hunk 2]
```diff
@@ -19,7 +19,7 @@ $ChocoCacheDir = 'C:\ChocoCache'
 $Qt32Version = "5.15.2"
 $Qt64Version = "5.15.2"
 $AqtinstallVersion = "2.1.0"
-$JackVersion = "1.9.17"
+$JackVersion = "1.9.21.20220626"
```
Comment (pljones) [inline .github/autobuild/windows.ps1:22]: Where did this version number come from?  The PR says just 1.9.21 without explanation.  This looks like a patch or nightly or some non-standard release.  We should have a good reason for that.

Comment (hoffie) [reply .github/autobuild/windows.ps1:22]: It's "vanilla jack 1.9.21 as packaged for choco as of 20220626". The initial jack 1.9.21 choco package release did not work as expected (see above), which is why the released an updated package with this version.

This information is in a [collapsed conversation thread](https://github.com/jamulussoftware/jamulus/pull/2665#discussion_r913418469) above (I've just uncollapsed it due to its relevance).

The end-user relevant information is still that JACK has been updated to 1.9.21. Maybe we could have a code comment. Personally, I think this PR is sufficient as there is nothing special to remember when doing the next bump or something.

Comment (pljones) [reply .github/autobuild/windows.ps1:22]: No, not important - just I missed the collapse thread.  Thanks.

**Review by pljones — APPROVED** [review]

**Pull request merged**